spark-scala/src/test/scala/com/matthewrathbone/spark/SparkJoinsTest.scala

```diff
@@ -0,0 +1,36 @@
+package test.scala.com.matthewrathbone.spark
+
+import org.scalatest.junit.AssertionsForJUnit
+import org.junit.Before
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.junit.Test
+import scala.collection.Map
+import org.apache.spark.SparkContext._
+
+class SparkJoinsTest extends AssertionsForJUnit {
+  
+  var sc: SparkContext = _
+  
+  @Before def initialize() {
+	 val conf = new SparkConf().setAppName("SparkJoins").setMaster("local")
+	 sc = new SparkContext(conf)
+  }
+  
+  @Test def sortByKey() {
+    val transactions: List[Tuple2[Int, Int]] = List(new Tuple2[Int, Int](1, 1), new Tuple2[Int, Int](2, 1),
```

Comment:
Correct! This is two things:

1. Tuple2 is a case class. Case classes do not require the `new` keyword, as there is a special apply method generated at compile time for each one.

2. There is a shortcut of just using parenthesis like `(a, b, c)` built into scala for Tuples so you don't always have to specify the type, just to make code cleaner. You can also use this as a type signature:

```scala
def someFunction(): (String, String) = {} // same as Tuple2[String, String]
def otherFunction(): (String, String, Int) = {} // same as Tuple3[String, String, Int]

```